mantidproject/mantid
# Check unsaved changes in table before closing ISIS Reflectometry Polref Interface

To check the table for unsaved changes when the user wishes to close the interface I needed to override the `closeEvent()` method in QWidget. This PR aims to allow the user the option to save their table out to a TBL file if there are unsaved changes in the processing table upon exiting the interface.

**To test:**
- Code Review
- Open ISIS Reflectometry (Polref) interface (`Interfaces > Reflectometry > ISIS Reflectometry (Polref)`)
- Close the interface and ensure no unsaved check is performed.
- Open the interface again
- Edit the processing table
- Close the interface
- Ensure that an unsaved check is performed before closing.

Fixes #16249

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
